### PR TITLE
Machine job feature bug fix

### LINKED ISSUE #155: Cleanup machine and job features templates
Use of metaconfig etc. as discussed in #152.

### PR DESCRIPTION
**Pull request by Pansanel**
The changes are based on https://gitlab.cern.ch/machinejobfeatures/mjf-scripts.
Fix #155

### CODE REVIEW

File: personality/wn/machine-features.pan [hunk 1]
```diff
@@ -9,6 +9,26 @@ variable MACHINE_FEATURES_LIST ?= list(
     'total_cpu',
 );
 
+variable MACHINE_FEATURES_SCRIPT ?= 'mjf';
```
Comment (jrha) [inline personality/wn/machine-features.pan]: Does this really need to be customisable? The following code would be much simpler if it wasn't.

File: personality/wn/machine-features.pan [hunk 2]
```diff
@@ -9,6 +9,26 @@ variable MACHINE_FEATURES_LIST ?= list(
...
+        'perms', '0755',
+        'owner', 'root',
+        'restart', MACHINE_FEATURES_STARTUP
+    );
+    SELF;
+};
+
+'/software/components/chkconfig' = {
+    if ( index('filecopy', SELF['dependencies']['pre']) < 0 ) {
+        SELF['dependencies']['pre'][length(SELF['dependencies']['pre'])] = 'filecopy';
```
Comment (jrha) [inline personality/wn/machine-features.pan]: Would this be easier to understand?
```pan
SELF['dependencies']['pre'] = append('filecopy');
```

Or you could skip the check and just do:
```pan
prefix '/software/components/chkconfig';
'dependencies/pre' ?= list();
'dependencies/pre' = merge(SELF, list('filecopy'));
'service' ?= dict();
'service' = merge(SELF, dict(
    MACHINE_FEATURES_SCRIPT, 'on',
));
```

File: personality/wn/machine-features.pan (hunk 1 above)
Comment (Pansanel) [reply personality/wn/machine-features.pan]: Yes, it should be customizable. The script can be modified to fit the configuration of the site and the batch scheduler.

File: personality/wn/machine-features.pan (hunk 2 above)
Comment (Pansanel) [reply personality/wn/machine-features.pan]: The first option gives the following error:
```
referenced variable of type nlist is not a list
```
I have other error with the second proposition (I prefer the first).

File: personality/wn/machine-features.pan (hunk 1 above)
Comment (jrha) [reply personality/wn/machine-features.pan]: Ok, thanks.

**Pull request merged**